openwisp_monitoring/monitoring/migrations/0013_delete_traffic_metrics.py

```diff
@@ -0,0 +1,20 @@
+from django.db import migrations
+
+
+def delete_traffic_metrics(apps, schema_editor):
+    Metric = apps.get_model('monitoring', 'Metric')
+    Metric.objects.filter(field_name='rx_bytes').delete()
```

Comment:
can we rename these instead of deleting them?

We have threshold configuration set for some metrics, and if we delete these we'll lose them.

Reply:
Is AlertSettings being used along with traffic metrics? (it's not the case atleast by default).
So if AlertSettings have been created for traffic metrics, there shall be a small problem. Currently, AlertSettings defaults to `field_name` for the value and there is no way to pass `extra_fields` to AlertSettings I think. A limitation which can be overcomed, so let me know if I should go ahead with it.

Reply:
There's no default alert settings but they have been added in some cases.
Moreover, if a device is temporarily down, the metrics and graphs will not show up after this migration, which will leave the user thinking there's a bug.
Whenever we introduce changes, we must make sure it doesn't affect negatively the people who are already using the software.
You can manipulate the existing metrics by looping over them right?
And the ones that are not needed anymore can be deleted as you're doing here. It shouldn't be a big change, isn't it?

Reply:
The small problem here is, there are 2 metrics, 2 AlertSettings. Now we can go ahead and merge the two metrics but we can't do so with AlertSettings (in the current form). The reason being AlertSettings currently supports only one threshold value. Since, metric supports multiple fields, it would be better if AlertSettings can also do the same.

**Proposal**: Let's have a TextField called additional_thresholds (this field shall hold the name and value of the extra_field's threshold as a list of dict).
```python
m.alertsettings.additional_fields = [{'tx_bytes': <threshold_value>}] # rx_bytes is covered by default, being the field_name
```
We can modify the change form to make this UI friendly and we aren't modifying the current approach, so it should be backward compatible. Metric configuration PR won't be much helpful here but yes we can make use of `metric.related_fields` to verify the fields entered in `alertsettings.additional_fields` exist. Let me know, if I should go ahead with this solution.

Reply:
I like it but I think it's less confusing if we add an extra metric_name field and keep having one AlertSettings per threshold. Otherwise you need also to define additional_operators, right?

This way you can get a notification only when rx_bytes goes above X and you don't really mind about tx_bytes.

Reply:
@PabloCastellano, @nemesisdesign let me know if the current iteration implements your suggestions well.